core/templates/pages/admin-page/roles-tab/admin-roles-tab.directive.ts

```diff
@@ -84,29 +93,223 @@ angular.module('oppia').directive('adminRolesTab', [
             role: values.newRole,
             username: values.username,
             topic_id: values.topicId
-          }).then(function() {
+          }).then(function(response) {
             ctrl.setStatusMessage(
-              'Role of ' + values.username +
-              ' successfully updated to ' + values.newRole);
-            ctrl.updateFormValues.username = '';
-            ctrl.updateFormValues.newRole = '';
-            ctrl.updateFormValues.topicId = '';
-          }, function(errorResponse) {
+              'Role of ' + values.username + ' successfully updated to ' +
+              values.newRole);
+            refreshFormData();
+          }, errorHandler);
+          AdminTaskManagerService.finishTask();
+        };
+
+        ctrl.submitAddCommunityReviewerForm = function(values) {
```

Comment:
"values" is quite a vague name. Can you make it more specific, e.g. formResponse?

Ditto below and elsewhere.